### ISSUE TITLE
Serenity-JS/rest: Axios version issue when using an `AxiosInstance` with latest axios version

### ISSUE THREAD

**Issue by viper3400** [serenity-js/serenity-js#1223]
I had to adjust the timeout of my `axios` instance as mentioned in https://github.com/serenity-js/serenity-js/issues/1196 described in the second example [here](https://serenity-js.org/modules/rest/class/src/screenplay/abilities/CallAnApi.ts~CallAnApi.html)

To do so, I run `npm install axios` which added the latest axios version, at the time of writing this was 0.27.4.

It results in this error:
```
Argument of type 'import("C:/.../e2e-serenityjs/node_modules/axios/index").AxiosInstance' is not assignable to parameter of type 'import("C:/.../e2e-serenityjs/node_modules/@serenity-js/rest/node_modules/axios/index").AxiosInstance'.
  The types of 'defaults.method' are incompatible between these types.
    Type 'string' is not assignable to type 'Method'.ts(2345)
```

`@serenity-js/rest` still uses an [axios version 0.21.4](https://github.com/serenity-js/serenity-js/blob/8bd28e99ef21ce0cf44e9c95e64761b0a89196fa/packages/rest/package.json#L52)

I downgraded the axios package in my project to this version - and it was working again.

I did no further analysis so far, so I'm not sure if this is just because of the general version mismatch or if breaking changes where introduced in axios meanwhile.

**Comment by jan-molak** [comment]
Hey @viper3400, `@serenity-js/rest` is purposefully depending on an older version of Axios, since the newer ones have this issue - https://github.com/axios/axios/issues/4124

A fix to that has been merged to Axios `master`, but it looks like it's never been released - https://github.com/axios/axios/pull/4229

Interestingly enough, based on https://github.com/axios/axios/issues/4304#issuecomment-1015244291, I see that @alecgibson has updated `@types/node` to introduce `AbortController` and `AbortSignal` - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58270, so we might be able to upgrade our dependency on Axios even without https://github.com/axios/axios/pull/4229

**Comment by jan-molak** [comment]
Released in 3.0.0-rc.19
